pandas/core/indexes/multi.py

```diff
@@ -2791,8 +2791,14 @@ def difference(self, other, sort=True):
                               labels=[[]] * self.nlevels,
                               names=result_names, verify_integrity=False)
 
-        difference = set(self._ndarray_values) - set(other._ndarray_values)
+        this = self._get_unique_index()
```

Comment:
The old way of doing this using ``set`` did not preserve the original order so I took this code from the ``difference`` method in ``pandas/core/indexes/base.py``:
https://github.com/pandas-dev/pandas/blob/145c2275e3560edc30ff418a57df67ba3c4c30d6/pandas/core/indexes/base.py#L2950-L2957